framework/src/modules/interoperability/base_interoperability_internal_methods.ts

```diff
@@ -479,6 +479,12 @@ export abstract class BaseInteroperabilityInternalMethod extends BaseInternalMet
 			throw new Error('Sending chain cannot be the receiving chain.');
 		}
 		// receivingChainID must correspond to a live chain.
+		// `timestamp` is only used in MainchainInteroperabilityInternalMethod::isLive
+		// but not in SidechainInteroperabilityInternalMethod::isLive
+		const mainchainID = getMainchainID(ownChainAccount.chainID);
+		if (ownChainAccount.chainID.equals(mainchainID) && !timestamp) {
+			throw new Error('Timestamp must be provided in mainchain context.');
+		}
 		const isReceivingChainLive = await this.isLive(context, receivingChainID, timestamp);
```

Comment:
maybe its better if we pass timestamp only for mainchain and not for sidechain 
On mainchain, `this.isLive(context, receivingChainID, timestamp);`
On sidechain `this.isLive(context, receivingChainID);`